src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -256,15 +256,28 @@ internal ClrRuntime CreateRuntime(ClrInfo clrInfo)
         {
             if (clrInfo == null) throw new ArgumentNullException(nameof(clrInfo));
 
-            string dac = clrInfo.LocalMatchingDac;
-            if (dac != null && !File.Exists(dac))
-                dac = null;
+            string localDacFileName = ClrInfoProvider.GetDacFileName(clrInfo.Flavor, clrInfo.Platform);
+            string localDacFilePath = Path.Combine(Path.GetDirectoryName(clrInfo.ModuleInfo.FileName), localDacFileName);
 
-            if (dac == null)
-                dac = SymbolLocator.FindBinary(clrInfo.DacInfo);
+            if (clrInfo.Platform == Platform.Linux) //strange logic here (originally from DataTargetImpl.InitVersions)
+            {
+                if (!File.Exists(localDacFilePath))
+                    localDacFilePath = localDacFileName;
+            }
+
+            string dac;
+            if (File.Exists(localDacFilePath) && PlatformFunctions.IsEqualFileVersion(localDacFilePath, clrInfo.Version))
+            {
+                dac = localDacFilePath;
+            }
+            else
+            {
+                string requestFileName = ClrInfoProvider.GetDacRequestFileName(clrInfo.Flavor, Architecture, Architecture, clrInfo.Version, clrInfo.Platform);
+                dac = SymbolLocator.FindBinary(requestFileName, 0, 0, false);
```

Comment:
Why do you request the dac with properties `0, 0`?  This would not find the correct Dac.  The purpose of DacInfo is to track the dac long-name as well as the filesize/timestamp of clr.dll itself (these are the properties required to request the dac).

Reply:
thanks for the details, looks like a misinterpretation from my side